receiver/zipkin/trace_receiver.go

```diff
@@ -404,30 +405,42 @@ func nodeFromZipkinEndpoints(zs *zipkinmodel.SpanModel) *commonpb.Node {
 		//      "zipkin.remoteEndpoint.port": "9000"
 		//      "zipkin.remoteEndpoint.serviceName": "backend",
 		// }
-		node.Attributes = zipkinEndpointIntoAttributes(rep, node.Attributes, func(s string) string {
-			return "zipkin.remoteEndpoint." + s
-		})
+		node.Attributes = zipkinEndpointIntoAttributes(rep, node.Attributes, isRemoteEndpoint)
 	}
 	return node
 }
 
+type zipkinDirection bool
+
+const (
+	isLocalEndpoint  zipkinDirection = true
+	isRemoteEndpoint zipkinDirection = false
+)
+
 var blankIP net.IP
 
-func zipkinEndpointIntoAttributes(ep *zipkinmodel.Endpoint, into map[string]string, prefixFunc func(string) string) map[string]string {
+func zipkinEndpointIntoAttributes(ep *zipkinmodel.Endpoint, into map[string]string, endpointType zipkinDirection) map[string]string {
 	if into == nil {
 		into = make(map[string]string)
```

Comment:
I considered making this allocation conditional on having data to insert but then the change is going to be risky, so doing only the safe change in this PR.